APMrover2/AP_Arming.cpp

```diff
@@ -185,3 +186,13 @@ bool AP_Arming_Rover::parameter_checks(bool report)
     return true;
 }
 
+// check if arming allowed from this mode
+bool AP_Arming_Rover::mode_checks(bool report)
+{
+    //display failure if arming is not allowed
+    if (!rover.control_mode->allows_arming_from_transmitter()) {
```

Comment:
ideally, we could transfert all this into AP_Arming, as it will be the same for all vehicle

Reply:
We would need either a common base class for the control mode or some sort of callback.  In either case, too much work for this PR IMO.